mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/ExtensionModelFactory.java

```diff
@@ -297,18 +305,71 @@ private SourceModel toMessageSource(SourceDeclaration declaration) {
     }
 
     private OperationModel toOperation(OperationDeclaration declaration) {
-      return fromCache(declaration, () -> new ImmutableOperationModel(declaration.getName(),
-                                                                      declaration.getDescription(),
-                                                                      toParameterGroups(declaration.getParameterGroups()),
-                                                                      toOutputModel(declaration.getOutput()),
-                                                                      toOutputModel(declaration.getOutputAttributes()),
-                                                                      declaration.isBlocking(),
-                                                                      declaration.getExecutionType(),
-                                                                      declaration.isRequiresConnection(),
-                                                                      declaration.isTransactional(),
-                                                                      declaration.getDisplayModel(),
-                                                                      declaration.getErrorModels(),
-                                                                      declaration.getModelProperties()));
+      return fromCache(declaration, () -> {
+        OperationModel operation;
+        if (declaration instanceof ScopeDeclaration) {
```

Comment:
There's something wrong here. I don't think ScopeModel should have an IS relationship with OperationModel. It may be that today they shared the same fields/behaviour but they represent different concepts that in the future will probably differ.  Seems we have to create an abstraction between OperationModel, ScopeModel and RouterModel and all of the possible Processors (which are not MessageSources).